cwms-data-api/src/main/java/cwms/cda/api/BasinController.java

```diff
@@ -312,19 +309,13 @@ public void create(@NotNull Context ctx) {
     @Override
     public void delete(@NotNull Context ctx, @NotNull String name) {
         DSLContext dsl = getDslContext(ctx);
-        String deleteMethod = ctx.queryParam(DELETE_MODE);
+        JooqDao.DeleteMethod deleteMethod = requiredParamAs(ctx, METHOD, JooqDao.DeleteMethod.class);
         cwms.cda.data.dao.basin.BasinDao basinDao = new cwms.cda.data.dao.basin.BasinDao(dsl);
         CwmsId basinId = new CwmsId.Builder()
                 .withName(name)
                 .withOfficeId(ctx.queryParam(OFFICE))
                 .build();
-        cwms.cda.data.dto.basin.Basin retBasin = basinDao.getBasin(basinId, "EN");
-        if (retBasin == null) {
-            CdaError error = new CdaError("No matching basin " + name);
-            ctx.status(HttpServletResponse.SC_NOT_FOUND).json(error);
-            return;
-        }
-        basinDao.deleteBasin(basinId, deleteMethod);
+        basinDao.deleteBasin(basinId, deleteMethod.getRule().getRule());
```

Comment:
getRule().getRule() -> Worth renaming one of those for clarity?

Reply:
updated to use DeleteRule in the DAO for consistency